src/test/java/it/gov/pagopa/arc/security/JwtAuthenticationFilterTest.java

```diff
@@ -0,0 +1,105 @@
+  private MemoryAppender memoryAppender;
+
+  @BeforeEach
+  public void setUp() {
+    SecurityContextHolder.clearContext();
+    ch.qos.logback.classic.Logger logger = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger("it.gov.pagopa.arc.security.JwtAuthenticationFilter");
+    memoryAppender = new MemoryAppender();
+    memoryAppender.setContext((LoggerContext) LoggerFactory.getILoggerFactory());
+    logger.setLevel(Level.ERROR);
+    logger.addAppender(memoryAppender);
+    memoryAppender.start();
+  }
+  @AfterEach
+  public void clearContext() {
+    SecurityContextHolder.clearContext();
+  }
+
+  @Test
+  void givenValidAuthenticationTokenThenVerifyThatSecurityContextIsCreated()
+      throws ServletException, IOException {
+    IamUserInfoDTO userInfoDTO = IamUserInfoDTOFaker.mockInstance();
+    Mockito.when(tokenStoreService.getUserInfo(sampleJwt)).thenReturn(
+        Optional.of(userInfoDTO));
+
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.addHeader("Authorization","Bearer "+sampleJwt);
+    MockHttpServletResponse response = new MockHttpServletResponse();
+    jwtAuthenticationFilter.doFilterInternal(request,response,new MockFilterChain());
+
+    Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+    Assertions.assertNotNull(principal);
+
+    Assertions.assertEquals(userInfoDTO.getUserId(),((IamUserInfoDTO)principal).getUserId());
+    Assertions.assertEquals(userInfoDTO.getName(),((IamUserInfoDTO)principal).getName());
+    Assertions.assertEquals(userInfoDTO.getFamilyName(),((IamUserInfoDTO)principal).getFamilyName());
+    Assertions.assertEquals(userInfoDTO.getEmail(),((IamUserInfoDTO)principal).getEmail());
+    Assertions.assertEquals(userInfoDTO.getFiscalCode(),((IamUserInfoDTO)principal).getFiscalCode());
+  }
+
+  @Test
+  void givenInValidAuthenticationTokenThenVerifyThatSecurityContextIsCreated()
+      throws ServletException, IOException {
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.addHeader("Authorization","Bearer ");
+    MockHttpServletResponse response = new MockHttpServletResponse();
+    jwtAuthenticationFilter.doFilterInternal(request,response,new MockFilterChain());
+    Assertions.assertNull(SecurityContextHolder.getContext().getAuthentication());
+  }
+
+  @Test
+  void givenUnknownErrorThenThrowException() throws ServletException, IOException {
+
+    //given
+    MockHttpServletRequest request = new MockHttpServletRequest();
+    request.addHeader("Authorization","Bearer "+sampleJwt);
+    MockHttpServletResponse response = new MockHttpServletResponse();
+
+    //when
+    Mockito.when(tokenStoreService.getUserInfo(sampleJwt)).thenThrow(new InvalidTokenException("Invalid token"));
+    jwtAuthenticationFilter.doFilterInternal(request,response,new MockFilterChain());
+
+    //then
+    Assertions.assertTrue(memoryAppender.getLoggedEvents().get(0).getFormattedMessage()
```

Comment:
assertion on logs is usually very weak. A log may change anytime and if a test refactor is needed everytime it may cause a maintainability issue

Reply:
yes this is clear, will remove test on log